daffodil-runtime1-unparser/src/main/scala/org/apache/daffodil/unparsers/runtime1/BinaryNumberUnparsers.scala

```diff
@@ -78,17 +76,61 @@ abstract class BinaryNumberBaseUnparser(override val context: ElementRuntimeData
 
 }
 
-abstract class BinaryIntegerBaseUnparser(e: ElementRuntimeData, signed: Boolean)
+abstract class BinaryIntegerBaseUnparser(e: ElementRuntimeData)
   extends BinaryNumberBaseUnparser(e) {
 
+  private val primNumeric = e.optPrimType.get.asInstanceOf[NodeInfo.PrimType.PrimNumeric]
+
   override def putNumber(
     dos: DataOutputStream,
     value: JNumber,
     nBits: Int,
     finfo: FormatInfo
   ): Boolean = {
+    val state = finfo.asInstanceOf[UState]
+    if (primNumeric.minWidth.isDefined) {
+      val isSigned = primNumeric.isSigned
+      val signedStr = if (isSigned) "signed" else "unsigned"
+      val minWidth = primNumeric.minWidth.get
+      if(nBits < minWidth) {
+        val outOfRangeFmtStr =
+          "Minimum length for a %s binary integer is %d bit(s), number of bits %d out of range. " +
+            "An unsigned integer with length 1 bit could be used instead."
+        if (isSigned && state.tunable.allowSignedIntegerLength1Bit) {
+          state.SDW(
+            WarnID.SignedBinaryIntegerLength1Bit,
+            outOfRangeFmtStr,
+            signedStr,
+            minWidth,
+            nBits
+          )
+          return false
```

Comment:
Is this `return false` correct? If we are only warning then should we still unparse something? For example, if nBits is 1 and we are signed but allowSignedIntegerLength1Bit, then we should output 1 bit. Seems like we want to warn but then drop through and still put a number.

Reply:
Hmm, so I have a test that exercises it, and we go from an infoset containing -1 to unparsing 1, since it is only 1 bit. Is that ok?

Reply:
I think that's probably correct? If we think about the min/max values of a 2's complement value, the min value is given as -(2<sup>n-1</sup>) and the max value is (2<sup>n-1</sup>)-1, where n is the number of bits. That gives us a range of -1 to 0. So -1 unparsing to a single 1 bit seems reasonable. Hopefully our parse logic is consistent with that when this tunabled is enabled.

Reply:
So it looks like we parse data that is -1 to an infoset containing 0, which I think we should be unparsing to 0 instead of 1 here since the max value is 0?

Reply:
> So it looks like we parse data that is -1 to an infoset containing 0, which I think we should be unparsing to 0 instead of 1 here since the max value is 0?

What does "parse data that is -1" mean? How can the data aka bits be -1. 

Let's use this terminology: data means the representation, aka the "bit stream". 
The value, logical value, or infoset value is a number with no concept of representation. 

Unparse logical value -1 should give a single '1' bit in the data stream. Unparse 0 should give a single '0' bit in the data stream. Parsing is symmetric.

Reply:
Hmm, seems like we have a bug in that parse and unparse are not symmetric.

For parse,
* 0x1 parses to 1
* 0x0 parses to 0

For unparse,
* 1 unparses to 0x0
* 0 unparse to 0x0
* -1 unparses to 0x1

Since parse will never create a -1 in the infoset, it means it will always unparse to 0 unless someone modifies the infoset, which is unlikely.

Considering this, I wonder if we should remove the new tunable and just drop support for signed 1-bit integers completely? If someone was already using this and roundtripping data, things were going to be subtly buggy.  So maybe changing this to an error is going to make it more obvious that things are broken and people can fix their schema?

And maybe we just add a deprecation/compatibility note that signed 1-bit integers were broken and shouldn't be supported anyways, so they are removed.

Reply:
Truncation of a number because it doesn't fit in the available space is supposed to be an unparse error.